### PR TITLE
KTOR-6745 Add a new tutorial 'Full Stack development with Kotlin Multiplatform'

### PR DESCRIPTION
**Pull request by vnikolova**
[KTOR-6745](https://youtrack.jetbrains.com/issue/KTOR-6745)
[Preview on staging](https://ktor.shared.aws.intellij.net/docs/full-stack-development-with-kotlin-multiplatform.html)

- proofread and added new content from Garth's article 'Full Stack development with Kotlin Multiplatform'
- added screenshots and gifs for light and dark theme
- added and referenced code example
- added link and meta descriptions

to do:
- [x] replace the GIFs in "create a server" and "run the service" sections
- [x] display procedures in side navigation
- [x] add a TLDR element to the top with the code example and libraries used
- [x] include links in the "Next steps" section

### CODE REVIEW

File: codeSnippets/snippets/tutorial-full-stack-task-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Task Manager Application with Kotlin Multiplatform
+
+An example Task Manager application built with Ktor and Kotlin Multiplatform
+by following the steps in the [Full-stack development with Kotlin Multiplatform
+](https://ktor.io/docs/full-stack-development-with-kotlin-multiplatform.html) tutorial.
+
+## Run the server
+
+1. In a terminal window run the following command from the root directory to build an executable that includes both the
```
Comment (Stexxe) [inline codeSnippets/snippets/tutorial-full-stack-task-manager/README.md:9]: Strictly speaking It's not executable

File: codeSnippets/snippets/tutorial-full-stack-task-manager/composeApp/src/commonMain/kotlin/network/TaskApi.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        install(ContentNegotiation) {
+            json(Json {
+                encodeDefaults = true
+                isLenient = true
+                coerceInputValues = true
+                ignoreUnknownKeys = true
+            })
+        }
+        defaultRequest {
+            host = "1.2.3.4"
```
Comment (Stexxe) [inline codeSnippets/snippets/tutorial-full-stack-task-manager/composeApp/src/commonMain/kotlin/network/TaskApi.kt:22]: Why this host?

File: codeSnippets/snippets/tutorial-full-stack-task-manager/shared/src/commonMain/kotlin/Greeting.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Greeting {
```
Comment (Stexxe) [inline codeSnippets/snippets/tutorial-full-stack-task-manager/shared/src/commonMain/kotlin/Greeting.kt:1]: Is this file used?

File: codeSnippets/snippets/tutorial-full-stack-task-manager/shared/src/commonMain/kotlin/Platform.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+interface Platform {
```
Comment (Stexxe) [inline codeSnippets/snippets/tutorial-full-stack-task-manager/shared/src/commonMain/kotlin/Platform.kt:1]: Is this file used?

**Review by Stexxe — APPROVED** [review]

**Review by e5l — APPROVED** [review]
Could we use fleet instead of android studio?

File: codeSnippets/snippets/tutorial-full-stack-task-manager/composeApp/src/commonMain/kotlin/network/TaskApi.kt (hunk 2 above)
Comment (vnikolova) [reply codeSnippets/snippets/tutorial-full-stack-task-manager/composeApp/src/commonMain/kotlin/network/TaskApi.kt:22]: It's just a placeholder. It should be replaced with the user's IP address (as mentioned in Step 4 in [Create the client](https://ktor.shared.aws.intellij.net/docs/full-stack-development-with-kotlin-multiplatform.html#create-client)).

File: codeSnippets/snippets/tutorial-full-stack-task-manager/shared/src/commonMain/kotlin/Greeting.kt (hunk 3 above)
Comment (vnikolova) [reply codeSnippets/snippets/tutorial-full-stack-task-manager/shared/src/commonMain/kotlin/Greeting.kt:1]: Not in the final application but we explain its function in the [Examine the project](https://ktor.shared.aws.intellij.net/docs/full-stack-development-with-kotlin-multiplatform.html#examine-project) section.

File: codeSnippets/snippets/tutorial-full-stack-task-manager/shared/src/commonMain/kotlin/Platform.kt (hunk 4 above)
Comment (vnikolova) [reply codeSnippets/snippets/tutorial-full-stack-task-manager/shared/src/commonMain/kotlin/Platform.kt:1]: Yes, it's also referenced in [Examine the project](https://ktor.shared.aws.intellij.net/docs/full-stack-development-with-kotlin-multiplatform.html#examine-project)

**Comment by e5l** [comment]
It has preview feature for the compose multiplatform and allows to run and debug both platforms in a single tool

**Pull request merged**